power_grid_model_c/power_grid_model/include/power_grid_model/optimizer/tap_position_optimizer.hpp

```diff
@@ -10,21 +10,30 @@
 #include "../common/enum.hpp"
 
 #include <boost/graph/compressed_sparse_row_graph.hpp>
+#include <functional>
+#include <queue>
+#include <vector>
 
 namespace power_grid_model::optimizer {
 namespace tap_position_optimizer {
 
 namespace detail = power_grid_model::optimizer::detail;
 
-using TrafoGraphIdx = size_t;
+using TrafoGraphIdx = size_t; // Questionable: should we simply use Idx?
```

Comment:
i'm ok with it - not sure what boost uses internally, but it's probably a good idea to stick to their internal data type if they are sign-sensitive